### PR TITLE
Replace opt() for --dry/--force with properties

### PR DESCRIPTION
**Pull request by happz**
Better type checking, and these two options are the only two where parent's flag wins.

### CODE REVIEW

**Review by KwisatzHaderach — APPROVED** [review]

File: tmt/utils.py [hunk 1]
```diff
@@ -1072,6 +1069,45 @@ def quietness(self) -> bool:
 
         return self._logger.quiet
 
+    # TODO: interestingly, the option has its own default, right? So why do we
+    # need a default of our own? Because sometimes commands have not been
+    # invoked, and there's not CLI invocation to ask for the default value.
```
Comment (psss) [inline tmt/utils.py:1074]:
```suggestion
    # invoked, and there's no CLI invocation to ask for the default value.
```

File: tmt/utils.py [hunk 2]
```diff
@@ -1072,6 +1069,45 @@ def quietness(self) -> bool:
...
+    # Maybe se should add some kind of "default invocation"...
```
Comment (psss) [inline tmt/utils.py:1075]:
```suggestion
    # Maybe we should add some kind of "default invocation"...
```

File: tmt/utils.py [hunk 3]
```diff
@@ -1072,6 +1069,45 @@ def quietness(self) -> bool:
...
+
+        invocation = self._inherited_cli_invocation
+
+        if invocation:
+            return invocation.options.get(option, default)
+
+        return default
+
+    @property
+    def is_dryrun(self) -> bool:
```
Comment (psss) [inline tmt/utils.py:1100]:
```suggestion
    def is_dry_run(self) -> bool:
```

Let's be consistent with separating words with underscores.

File: tmt/utils.py [hunk 4]
```diff
@@ -1072,6 +1069,45 @@ def quietness(self) -> bool:
...
+        """ Whether the current run is a dry-run """
+
+        return self._get_cli_flag('is_dryrun', 'dry', False)
+
+    @property
+    def is_forcedrun(self) -> bool:
```
Comment (psss) [inline tmt/utils.py:1106]:
```suggestion
    def is_forced_run(self) -> bool:
```

**Review by psss — DISMISSED** [review]
Looks good, just proposing consistent naming and found some typos. Seems simple enough to squeeze into `1.27`. @happz, shall we include it or would you like to keep it for `1.28`?

**Comment by happz** [comment]
> Looks good, just proposing consistent naming and found some typos. Seems simple enough to squeeze into `1.27`. @happz, shall we include it or would you like to keep it for `1.28`?

It's not a bug fix, I'm perfectly fine with merging this into 1.28.

File: tmt/utils.py (hunk 1 above)
Comment (happz) [reply tmt/utils.py]: Fixed in 7aa7a2ff

File: tmt/utils.py (hunk 2 above)
Comment (happz) [reply tmt/utils.py]: Fixed in 7aa7a2ff

File: tmt/utils.py (hunk 3 above)
Comment (happz) [reply tmt/utils.py]: Fixed in 7aa7a2ff

**Review by thrix — APPROVED** [review]
LGTM

**Review by psss — APPROVED** [review]

**Pull request merged**